<a href="https://colab.research.google.com/github/edyy-Z/Data-Analysis-Class-Works/blob/main/DataAssign_Lab_6_Sum24.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Lab 6

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [ ]:
df = pd.read_csv('sample_data/dtaset.csv')
df

make_model body_type  price               vat            km  \
0             Audi A1    Sedans  15770    VAT deductible  56013.000000   
1             Audi A1    Sedans  14500  Price negotiable  80000.000000   
2             Audi A1    Sedans  14640    VAT deductible  83450.000000   
3             Audi A1    Sedans  14500    VAT deductible  73000.000000   
4             Audi A1    Sedans  16790    VAT deductible  16200.000000   
...               ...       ...    ...               ...           ...   
15910  Renault Espace       Van  39950    VAT deductible   1647.362609   
15911  Renault Espace       Van  39885    VAT deductible   9900.000000   
15912  Renault Espace       Van  39875    VAT deductible     15.000000   
15913  Renault Espace       Van  39700    VAT deductible     10.000000   
15914  Renault Espace       Van  40999    VAT deductible   1647.362609   

                 Type     Fuel  Gears  \
0                Used   Diesel    7.0   
1                Used  Benzine    7.0   
2                Used   Diesel    7.0   
3                Used   Diesel    6.0   
4                Used   Diesel    7.0   
...               ...      ...    ...   
15910             New   Diesel    6.0   
15911            Used  Benzine    7.0   
15912  Pre-registered   Diesel    6.0   
15913  Pre-registered   Diesel    6.0   
15914   Demonstration  Benzine    6.0   

                                     Comfort_Convenience  \
0      Air conditioning,Armrest,Automatic climate con...   
1      Air conditioning,Automatic climate control,Hil...   
2      Air conditioning,Cruise control,Electrical sid...   
3      Air suspension,Armrest,Auxiliary heating,Elect...   
4      Air conditioning,Armrest,Automatic climate con...   
...                                                  ...   
15910  Air conditioning,Automatic climate control,Cru...   
15911  Air conditioning,Automatic climate control,Cru...   
15912  Air conditioning,Armrest,Automatic climate con...   
15913  Air conditioning,Automatic climate control,Cru...   
15914  Air conditioning,Automatic climate control,Cru...   

                                     Entertainment_Media  ... Previous_Owners  \
0      Bluetooth,Hands-free equipment,On-board comput...  ...             2.0   
1      Bluetooth,Hands-free equipment,On-board comput...  ...             1.0   
2                                  MP3,On-board computer  ...             1.0   
3      Bluetooth,CD player,Hands-free equipment,MP3,O...  ...             1.0   
4      Bluetooth,CD player,Hands-free equipment,MP3,O...  ...             1.0   
...                                                  ...  ...             ...   
15910  Bluetooth,Digital radio,Hands-free equipment,O...  ...             1.0   
15911  Bluetooth,Digital radio,Hands-free equipment,R...  ...             1.0   
15912  Bluetooth,Hands-free equipment,On-board comput...  ...             1.0   
15913                  Bluetooth,Digital radio,Radio,USB  ...             1.0   
15914                                                USB  ...             1.0   

       hp_kW  Inspection_new  Paint_Type    Upholstery_type  Gearing_Type  \
0       66.0               1    Metallic              Cloth     Automatic   
1      141.0               0    Metallic              Cloth     Automatic   
2       85.0               0    Metallic              Cloth     Automatic   
3       66.0               0    Metallic              Cloth     Automatic   
4       66.0               1    Metallic              Cloth     Automatic   
...      ...             ...         ...                ...           ...   
15910  147.0               0    Metallic  Part/Full Leather     Automatic   
15911  165.0               0    Metallic              Cloth     Automatic   
15912  146.0               1    Metallic  Part/Full Leather     Automatic   
15913  147.0               0    Metallic  Part/Full Leather     Automatic   
15914  165.0               1    Metallic  Part/Full Leather     Automatic   

      Disp

In [ ]:
df['price'].describe()

count    15915.000000
mean     18024.380584
std       7381.679318
min       4950.000000
25%      12850.000000
50%      16900.000000
75%      21900.000000
max      74600.000000
Name: price, dtype: float64

1. Run a multiple linear probability model (have at least 2 Xs in the model). Tell me how you think your independent variables will affect your dependent variable. Interpret your results. Were your expectations correct? Why or why not?

Car price is (price) the dependent variable. The data range from 4950 to 74600 with 4950 being cheap car and 74600 being expensive car. This has been recoded to binary so cheap car (4950-15000) is 0 and good economy (15000-74600) is 1. I believe that car price should be associated with a higher gear numbers (Gears), horsepower (hp_KW) , and a lower drived distance in km (km).

In [ ]:
sub = df.dropna(subset = ['price', 'Gears', 'hp_kW', 'km'])

In [ ]:
conditions = [(sub['price'] < 15000) , (sub['price'] > 15000)]
choices = [0, 1]
sub['good_car'] = np.select(conditions, choices, default=np.nan)

In [ ]:
pd.crosstab(index=sub["good_car"], columns="count")

col_0     count
good_car       
0.0        6026
1.0        9853

In [ ]:
lm1 = smf.ols(formula = 'good_car ~ Gears + hp_kW + km', data = sub).fit()
print (lm1.summary())

                            OLS Regression Results                            
Dep. Variable:               good_car   R-squared:                       0.447
Model:                            OLS   Adj. R-squared:                  0.447
Method:                 Least Squares   F-statistic:                     4284.
Date:                Tue, 10 Dec 2024   Prob (F-statistic):               0.00
Time:                        20:12:45   Log-Likelihood:                -6340.9
No. Observations:               15879   AIC:                         1.269e+04
Df Residuals:                   15875   BIC:                         1.272e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -1.1884      0.025    -48.389      0.0

Number of gears (Gears) is positively associated with the price of a car, and the association is statistically significant.

Horsepower (hp_KW) has a very small coefficient and is not statistically significant.

Drived distance (km) is negatively associated with the price, and the association is statistically significant.

2. Run a multiple (binary) logistic model. (It can be the same as the above LPM or a new model.) If it is a new model, tell me how you think your independent variables will affect your dependent variable. Interpret your results in the logit scale. Were your expectations correct? Why or why not?

In [ ]:
logit1 = sm.formula.logit(formula = 'good_car ~ Gears + hp_kW + km', data = sub).fit()
print (logit1.summary())

Optimization terminated successfully.
         Current function value: 0.354456
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:               good_car   No. Observations:                15879
Model:                          Logit   Df Residuals:                    15875
Method:                           MLE   Df Model:                            3
Date:                Tue, 10 Dec 2024   Pseudo R-squ.:                  0.4660
Time:                        20:12:45   Log-Likelihood:                -5628.4
converged:                       True   LL-Null:                       -10541.
Covariance Type:            nonrobust   LLR p-value:                     0.000
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -12.3707      0.235    -52.676      0.000     -12.831     -11.910
Gears          1.4247      0.

Logit Model shows the same results. Gears is positively associated with price of cars, and statistically significant. Hp_KW coefficient is very small and not statistically significant. Drived distance is negatively associated with the price, and the association is statistically significant.

3. Get odds ratios from your logit model in Question 2 and interpret some of them.

In [ ]:
np.exp(logit1.params)

Intercept    0.000004
Gears        4.156525
hp_kW        1.072250
km           0.999963
dtype: float64

For each one-unit increase in gears, the price of the car being considered huge increase by a factor of 316%. Hp_KW does affect the performance of the economy with one unit incease in horsepower, the price of the car would increase by 6.7% . For each one-unit increase in the dirved distance in km, the price of the car being considered a decrease by 0.04%.

4. Get predicted probabilities from your logit model in Question 2 for some constellations of X values and interpret the results.

In [ ]:
def logit2prob (logit):
    odds = np.exp(logit)
    prob = odds / (1 + odds)
    return(prob);

intercept = logit1.params.Intercept
b_Gears = logit1.params.Gears
b_hp_kW = logit1.params.hp_kW
b_km = logit1.params.km

Gears = 7, horsepower = 100, drived distacne = 1000

96% chance of economy being good

In [ ]:
logits_exh = intercept + (6 *b_Gears ) + (100 * b_hp_kW) + (1000 * b_km)
logit2prob(logits_exh)

0.9575529651038022

Gears = 4, horsepower = 50, drived distacne = 1000

0.4% chance of economy being good

In [ ]:
logits_exh = intercept + (4 *b_Gears ) + (50 * b_hp_kW) + (1000 * b_km)
logit2prob(logits_exh)

0.038375794199155565

Probability of car price being well with average values on the X

74% chance of the car being good

In [ ]:
logits_exh = intercept + (sub.Gears.mean() * b_Gears) + (sub.hp_kW.mean() * b_hp_kW) + (sub.km.mean() * b_km)
logit2prob(logits_exh)

0.7442909613492615